src/mainboard/clevo/adl-p/ramstage.c

```diff
@@ -76,17 +76,56 @@ static void set_camera_enablement(void)
 	system76_ec_smfi_cmd(CMD_CAMERA_ENABLEMENT_SET, sizeof(enabled) / sizeof(uint8_t), (uint8_t *)&enabled);
 }
 
+static void set_battery_thresholds(void)
+{
+	struct battery_config bat_cfg;
+
+	get_battery_config(&bat_cfg);
+
+	system76_ec_set_bat_threshold(BAT_THRESHOLD_START, bat_cfg.start_threshold);
+	system76_ec_set_bat_threshold(BAT_THRESHOLD_STOP, bat_cfg.stop_threshold);
+}
+
+static void set_power_on_ac(void)
+{
+	struct smfi_option_get_cmd {
+		uint8_t index;
+		uint8_t value;
+	} __packed cmd = {
+		OPT_POWER_ON_AC,
+		0
+	};
+
+	cmd.value = dasharo_get_power_on_after_fail();
+
+	system76_ec_smfi_cmd(CMD_OPTION_SET, sizeof(cmd) / sizeof(uint8_t), (uint8_t *)&cmd);
```

Comment:
Is it using the EC flash? If yes, then we should first check what is the current state from EC side and then if needed, issue CMD_OPTION_SET

Reply:
The check to avoid re-writing the same value to flash is done in the EC already: https://github.com/Dasharo/ec/pull/48/files#diff-603dee9b68e0fa20bb23508264232120edcd59f282a8dd59b999299c1a37f025R71